virtual-desktop/src/app/authentication-manager/login/login.component.ts

```diff
@@ -168,14 +172,33 @@ export class LoginComponent implements OnInit {
             let jsonMessage = JSON.parse(error);
             if (jsonMessage.categories) {
               let failedTypes = [];
+              let failedPlugins = new Set<string>();
               let keys = Object.keys(jsonMessage.categories);
               for (let i = 0; i < keys.length; i++) {
                 if (!jsonMessage.categories[keys[i]].success) {
                   failedTypes.push(keys[i]);
                 }
               }
-              this.errorMessage = this.translation.translate('AuthenticationFailed',
+              this.errorDetails = '';
+              let err;
+              for (let i = 0; i < failedTypes.length; i++) {
+                let plugins = Object.keys(jsonMessage.categories[failedTypes[i]].plugins);
+                for (let j = 0; j < plugins.length; j++) {
+                  err = jsonMessage.categories[failedTypes[i]].plugins[plugins[j]].error;
+                  if(err) {
+                    this.errorMessage = err.message;
+                    if(!failedPlugins.has(plugins[j])) {
+                      // Appending the error messages and corresponding unique plugin-ids that have errors
+                      this.errorDetails += `${plugins[j]}: ${this.errorMessage}.\n${err.body}\n`;
```

Comment:
Therefore these may also be undefined. In that case, is there something better you want to put other than
```
org.zowe.zlux.auth.example: undefined
undefined
```
Perhaps "No error message given" and "No error body given"